src/game/logic/object/experiencetracker.cpp

```diff
@@ -0,0 +1,181 @@
+/**
+ * @file
+ *
+ * @author Duncans_pumpkin
+ *
+ * @brief ExperienceTracker
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "experiencetracker.h"
+#include "gamelogic.h"
+#include "object.h"
+#include "xfer.h"
+
+// zh: 0x0061B790 wb: 0x008A35F9
+int32_t ExperienceTracker::Get_Experience_Value(const Object *killer) const
+{
+    if (killer->Get_Relationship(m_parent) == Relationship::ALLIES) {
+        return 0;
+    }
+
+    return m_parent->Get_Template()->Get_Experience_Value(m_currentLevel);
+}
+
+// zh: 0x0061B7D0 wb: 0x008A3637
+bool ExperienceTracker::Is_Trainable() const
+{
+    return m_parent->Get_Template()->Is_Trainable();
+}
+
+// zh: 0x0061B7E0 wb: 0x008A3654
+bool ExperienceTracker::Is_Accepting_Experience_Points() const
```

Comment:
This function is not used inside this class. So class user must call it before adding EXP. Is this proper?
`Can_Gain_Exp_For_Level` is also not used. But `Is_Trainable()` is. I am asking, because if checks are missing, then EXP may be added when it should not be. But it could be of course that class user calls these originally.

Reply:
Again just how vanilla is